client/gutenboarding/store/resolvers.ts

```diff
@@ -0,0 +1,31 @@
+/**
+ * External dependencies
+ */
+import { addQueryArgs } from '@wordpress/url';
+
+/**
+ * Internal dependencies
+ */
+import * as selectors from './selectors';
+import { apiFetch, dispatch } from '@wordpress/data-controls';
+import { STORE_KEY } from './constants';
+
+type ResolverParameters< S extends ( state: any, ...args: any ) => any > = S extends (
+	state: any,
+	...args: infer P
+) => any
+	? P
+	: never;
+
+/**
+ * Requests theme supports data from the index.
+ */
+export function* getVertical(
+	...[ search ]: ResolverParameters< typeof selectors[ 'getVertical' ] >
+) {
+	const url = `https://public-api.wordpress.com/wpcom/v2/verticals${ addQueryArgs( undefined, {
```

Comment:
I'm surprised to see the whole domain here instead of just API path.

Should API base be added by fetch middlewares instead?

Reply:
`apiFetch` really targets a WordPress environment and WordPress API calls. We don't really have one here. It might make sense to use `fetch` directly.

I'll explore that a bit…